aio/deploy/recommended.yaml

```diff
@@ -199,6 +199,19 @@ spec:
             # If not specified, Dashboard will attempt to auto discover the API server and connect
             # to it. Uncomment only if the default does not work.
             # - --apiserver-host=http://my-address:port
+          env:
```

Comment:
We can add that to head deployments, but I'd expect that for the recommended one people will use their own certificates. It would be even better for them to see some error and force them to use proper custom certificates instead of self-signed one.

Reply:
I removed the environment variables in recommended deployment YAMLs. 

I understand your concern on forcing user to create their own certificates, though I still believe most of them will create another self-signed certificate just to skip the "unskippable" auto-generated self-signed certificate, or even finding other ways to skip the error prompt by using CentOS, Firefox or modifying flags in Chrome.

We all know using self-signed certificate is not a best practice, but I don't think this should block people from deploying a quickstart environment. If forcing them to generate a proper custom certificate is a must, then I would recommend to not generate a self-signed certificate at all and let the program exit with error.

WDYT

Reply:
To be honest I haven't encountered any issues with an autogenerated certificate that we use right now. Anyway, even forcing someone to generate a custom self-signed certificate requires the user to be aware of what he is doing. It requires some knowledge, force you to read the documentation and all security issues that will come with it will be on the user.

Reply:
## Spec

- Operating System: Ubuntu 18.04.3 LTS
- Kubernetes version: v1.17.0
- Dashboard version: v2.0.0-rc3
- Browser: Chrome 79.0.3945.130 (Official Build) (64-bit) on Windows 10

## How to reproduce:

1. Install Kubernetes v1.17.0 with kubeadm and install network plugin
1. Install dashboard with command `kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0-rc3/aio/deploy/recommended.yaml`
1. Change the service type for `kubernetes-dashboard/kubernetes-dashboard` to `NodePort`
1. Connect to the dashboard through `https://NodeIP:NodePort`
![dashboard-cert-error](https://user-images.githubusercontent.com/13197632/73812755-e2d13d80-4818-11ea-8949-e6b5ac5c56e7.png)